extensions/sql-migration/src/api/azure.ts

```diff
@@ -510,16 +566,136 @@ export async function getLocationDisplayName(location: string): Promise<string>
+		kind: Kind.SQLDB,
+		validateIntegrationRuntimeOnline: testIrOnline,
+		sourceSqlConnection: {
+			testConnectivity: testSourceConnectivity,
+			dataSource: sourceServerName,
+			userName: migration._sqlServerUsername,
+			password: migration._sqlServerPassword,
+			authentication: migration._authenticationType,
+			trustServerCertificate: trustServerCertificate,
+			// encryptConnection: true,
+		},
+		targetSqlConnection: {
+			testConnectivity: testTargetConnectivity,
+			dataSource: targetDatabaseServer.properties.fullyQualifiedDomainName,
+			userName: migration._targetUserName,
+			password: migration._targetPassword,
+			encryptConnection: true,
+			trustServerCertificate: false,
+			authentication: MigrationSourceAuthenticationType.Sql,
+		}
+	};
+
+	const response = await api.makeAzureRestRequest(
+		account,
+		subscription,
+		path,
+		azurecore.HttpRequestMethod.POST,
+		requestBody,
+		true);
+
+	if (response.errors.length > 0) {
+		throw new Error(response.errors.map(e => e.message).join(','));
+	}
+	return response.response.data;
+}
+
+export async function validateIrDatabaseMigrationSettings(
+	migration: MigrationStateModel,
+	sourceServerName: string,
+	trustServerCertificate: boolean,
+	sourceDatabaseName: string,
+	networkShare: NetworkShare,
+	testIrOnline: boolean = true,
+	testSourceLocationConnectivity: boolean = true,
+	testSourceConnectivity: boolean = true,
+	testBlobConnectivity: boolean = true): Promise<ValdiateIrDatabaseMigrationResponse> {
+
+	const api = await getAzureCoreAPI();
+	const account = migration._azureAccount;
+	const subscription = migration._targetSubscription;
+	const serviceId = migration._sqlMigrationService?.id;
+	const path = encodeURI(`${serviceId}/validateIr?api-version=${DMSV2_API_VERSION}`);
+	const storage = await getStorageAccountAccessKeys(account, subscription, networkShare.storageAccount);
+
+	const requestBody: ValdiateIrDatabaseMigrationRequest = {
+		sourceDatabaseName: sourceDatabaseName ?? '',
+		kind: migration._targetType === MigrationTargetType.SQLMI
+			? Kind.SQLMI
+			: Kind.SQLVM,
+		validateIntegrationRuntimeOnline: testIrOnline,
+		backupConfiguration: {
+			sourceLocation: {
+				testConnectivity: testSourceLocationConnectivity,
+				fileShare: {
+					path: networkShare.networkShareLocation,
+					username: networkShare.windowsUser,
+					password: networkShare.password,
+				},
+			},
+			targetLocation: {
+				testConnectivity: testBlobConnectivity,
+				accountKey: storage?.keyName1,
+				storageAccountResourceId: networkShare.storageAccount?.id
+			},
+		},
+		sourceSqlConnection: {
+			testConnectivity: testSourceConnectivity,
+			dataSource: sourceServerName,
+			userName: migration._sqlServerUsername,
+			password: migration._sqlServerPassword,
+			trustServerCertificate: trustServerCertificate,
+			encryptConnection: true,
+			authentication: migration._authenticationType,
+		}
+	};
+
+	const response = await api.makeAzureRestRequest(
```

Comment:
can you please add the `host` parameter to the REST request, else the wrong endpoint will be used for non-public (e.g. USGov) users and the request will fail.

e.g.: https://github.com/microsoft/azuredatastudio/blob/16bea07a70195e68b63ee5017293c55374f8e772/extensions/sql-migration/src/api/azure.ts#L42-L43

Reply:
as well as anywhere else we make similar requests, in case I miss any of those too

Reply:
good catch, I did find 2